client/my-sites/site-settings/redirect-to/redirect-to.jsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import page from 'page';
+import { flowRight } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+import { isSiteAutomatedTransfer } from 'state/selectors';
+
+const redirectNonJetpack = ( WrappedComponent, redirectUrl ) => {
```

Comment:
This would actually require you to pass the `redirectUrl` unnecessarily to all other HoC that are in the `flowRight` call, for example `localize. I doubt you'd want to do that.

How about using another level of arrow functions here? It would make it more intuitive IMHO.

So instead of:

```
const redirectNonJetpack = ( WrappedComponent, redirectUrl ) => {
```

you would have:

```
const redirectNonJetpack = redirectUrl => WrappedComponent => {
```

and then in the `flowRight` sequence, instead of:

```
export default flowRight(
	localize,
	redirectNonJetpack
)( WrappedComponent, redirectUrl );
```

you would have:

```
export default flowRight(
	localize,
	redirectNonJetpack( redirectUrl )
)( WrappedComponent );
```

What do you think?

If you decide to go that way, don't forget to also update the HoC README.

Reply:
This is true, thanks! I did this initially and then rolled to the current version. As always, great catch 👍